mir_eval/transcription.py

```diff
@@ -0,0 +1,377 @@
+        and pitch matching are performed using ``<=`` (less than or equal). If
+        ``strict=True``, the threshold checks are performed using ``<`` (less
+        than).
+
+    Returns
+    -------
+    precision : float
+        The computed precision score
+    recall : float
+        The computed recall score
+    f_measure : float
+        The computed F-measure score
+    """
+    validate(ref_intervals, ref_pitches, est_intervals, est_pitches)
+    # When reference notes are empty, metrics are undefined, return 0's
+    if len(ref_pitches) == 0 or len(est_pitches) == 0:
+        return 0., 0., 0.
+
+    matching = match_notes(ref_intervals, ref_pitches, est_intervals,
+                           est_pitches, onset_tolerance=onset_tolerance,
+                           pitch_tolerance=pitch_tolerance,
+                           offset_ratio=offset_ratio,
+                           offset_min_tolerance=offset_min_tolerance,
+                           strict=strict)
+
+    precision = float(len(matching))/len(est_pitches)
+    recall = float(len(matching))/len(ref_pitches)
+    f_measure = util.f_measure(precision, recall)
+    return precision, recall, f_measure
+
+
+def evaluate(ref_intervals, ref_pitches, est_intervals, est_pitches, **kwargs):
+    """Compute all metrics for the given reference and estimated annotations.
+
+    Examples
+    --------
+    >>> ref_intervals, ref_pitches = mir_eval.io.load_valued_intervals(
+    ...     'reference.txt')
+    >>> est_intervals, est_pitches = mir_eval.io.load_valued_intervals(
+    ...     'estimate.txt')
+    >>> scores = mir_eval.transcription.evaluate(ref_intervals, ref_pitches,
+    ...     est_intervals, est_pitches)
+
+    Parameters
+    ----------
+    ref_intervals : np.ndarray, shape=(n,2)
+        Array of reference notes time intervals (onset and offset times)
+    ref_pitches: np.ndarray, shape=(n,)
+        Array of reference pitch values in Hertz
+    est_intervals : np.ndarray, shape=(m,2)
+        Array of estimated notes time intervals (onset and offset times)
+    est_pitches : np.ndarray, shape=(m,)
+        Array of estimated pitch values in Hertz
+    kwargs
+        Additional keyword arguments which will be passed to the
+        appropriate metric or preprocessing functions.
+
+    Returns
+    -------
+    scores : dict
+        Dictionary of scores, where the key is the metric name (str) and
+        the value is the (float) score achieved.
+    """
+    # Compute all the metrics
+    scores = collections.OrderedDict()
+
+    # Precision, recall and f-measure taking note offsets into account
+    offset_ratio = kwargs.get('offset_ratio', .2)
+    if offset_ratio is not None:
+        (scores['Precision'],
+         scores['Recall'],
+         scores['F-measure']) = util.filter_kwargs(
+            precision_recall_f1, ref_intervals, ref_pitches, est_intervals,
+            est_pitches, offset_ratio=offset_ratio, **kwargs)
```

Comment:
Sure, but 
```python
kwargs['offset_ratio'] = kwargs.get('offset_ratio', .2)
```
is equivalent to
```python
kwargs.setdefault('offset_ratio', 0.2)
```
and you could then chcek
```python
if kwargs['offset_ratio'] is not None:
```
instead of having multiple copies kicking around.

Splitting hairs here, but since it seems like there could be some shadow-copying going on, better to minimize the number of variables in play.